src/actionModifiers/ImportExportWithDatasets/importWithDatasetsModifier.js

```diff
@@ -164,9 +164,35 @@ function importWithDatasetsModifier(openmct) {
     }
 
     function getReferencedDatasets(json) {
-        const referencedDatasets = json[CONSTANTS.DATASETS_IDENTIFIER];
+        const openmct = json.openmct;
+        const referencedDatasets = new Set();
+
+        Object.values(openmct)
+            .forEach(object => object?.composition
```

Comment:
I think you want to move the conditional check after composition? we should always have an object here.

Reply:
😅